service/chartconfig/v1/resource/chart/delete.go

```diff
@@ -2,14 +2,70 @@ package chart
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/giantswarm/chart-operator/service/chartconfig/v1/key"
+	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
 )
 
 func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	chartState, err := toChartState(deleteChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if chartState.ReleaseName != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting release %s", chartState.ReleaseName))
+		release := key.ReleaseName(customObject)
+
+		err := r.helmClient.DeleteRelease(release)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting release %s", chartState.ReleaseName))
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("not deleting release %s", chartState.ReleaseName))
+	}
 	return nil
 }
 
 func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
+	delete, err := r.newDeleteChange(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	patch := framework.NewPatch()
+	patch.SetDeleteChange(delete)
+
+	return patch, nil
+}
+
+func (r *Resource) newDeleteChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	currentChartState, err := toChartState(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredChartState, err := toChartState(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding out if the %s chart has to be deleted", desiredChartState.ChartName))
+
+	if currentChartState.ChartName != "" && currentChartState.ChartName == desiredChartState.ChartName {
```

Comment:
No sure, whether we delete by `ChartName` or should use `ReleaseName` here as the helm client does.

Reply:
IMO this kind of condition is not very cool for our resource implementations. I saw this mostly in the aws-operator and in some places, like the cloudformation resource it was necessary to do it this way, but in most cases we only have to check if the current state is not nil or something. It does not really add up for me to check some arbitrary property of the state object. Like, why should a name be there but not some other property? Why don't we check the other properties? And so on.

Reply:
Alright will change that. :+1: 
Just out of interest: How do we now we are deleting the right chart? Thought we need `currentChartState.ChartName == desiredChartState.ChartName` to check that.